tools/scripts/ansible/templates/server.xml

```diff
@@ -84,15 +84,6 @@
          style configuration. When using the APR/native implementation, the
          OpenSSL style configuration is required as described in the APR/native
          documentation -->
-    <!--
-    <Connector address="127.0.0.1" port="8443" protocol="org.apache.coyote.http11.Http11NioProtocol"
-               maxThreads="150" SSLEnabled="true" scheme="https" secure="true"
-               clientAuth="false" sslProtocol="TLS" />
-    -->
-
-    <!-- Define an AJP 1.3 Connector on port 8009 -->
-    <Connector address="127.0.0.1" port="8009" protocol="AJP/1.3" redirectPort="8443" />
-
```

Comment:
avoid:
SEVERE [main] org.apache.catalina.util.LifecycleBase.handleSubClassException Failed to start component [Connector[AJP/1.3-8009]]
	org.apache.catalina.LifecycleException: Protocol handler start failed
		at org.apache.catalina.connector.Connector.startInternal(Connector.java:1067)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.StandardService.startInternal(StandardService.java:438)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
		at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)
	Caused by: java.lang.IllegalArgumentException: The AJP Connector is configured with secretRequired="true" but the secret attribute is either null or "". This combination is not valid.
		at org.apache.coyote.ajp.AbstractAjpProtocol.start(AbstractAjpProtocol.java:270)

Reply:
Fair enough… nginx recommends proxying with http instead of AJP: https://docs.nginx.com/nginx/deployment-guides/load-balance-third-party/apache-tomcat/

Reply:
I'm glad you knew that! I didn't, and I don't know for sure that we don't need AJP at all -- we'll see what happens on smoketest! BTW probably we don't need to supply an entire server.xml template; we could instead configure robertdebock.tomcat at a higher level

Reply:
AJP is another proxy protocol (non-HTTP). We don't use it.  "JK" is another protocol you might read of somewhere.

Tomcat turns it on by default, but since we don't need it, best to just throw it out

Reply:
See this _apache httpd_ example line showing how to connect to an AJP service:

https://httpd.apache.org/docs/2.4/mod/mod_proxy_ajp.html#usage